base_synchro/wizard/base_synchro.py

```diff
@@ -232,23 +230,24 @@ def data_transform(self, cr, uid, pool_src, pool_dest, object, data, action='u',
     # Synchronize these obejcts
     #
 
-
-    def upload_download(self, cr, uid, ids, context=None):
+    @api.multi
+    def upload_download(self):
         start_date = time.strftime('%Y-%m-%d, %Hh %Mm %Ss')
-        syn_obj = self.browse(cr, uid, ids, context=context)[0]
-        pool = pooler.get_pool(cr.dbname)
-        server = pool.get('base.synchro.server').browse(cr, uid, syn_obj.server_url.id, context=context)
+        syn_obj = self.browse(self.ids)[0]
+        pool = pooler.get_pool(self.env.cr.dbname)
+        server = self.env['base.synchro.server'].browse(syn_obj.server_url.id)
         for object in server.obj_ids:
             dt = time.strftime('%Y-%m-%d %H:%M:%S')
-            self.synchronize(cr, uid, server, object, context=context)
+            self.synchronize(server, object)
             if object.action == 'b':
                 time.sleep(1)
                 dt = time.strftime('%Y-%m-%d %H:%M:%S')
-            self.pool.get('base.synchro.obj').write(cr, uid, [object.id], {'synchronize_date': dt}, context=context)
+            self.env['base.synchro.obj'].write({'synchronize_date': dt})
         end_date = time.strftime('%Y-%m-%d, %Hh %Mm %Ss')
+        return {}
```

Comment:
Method will be returned here, so following code will never going to execute and user will not have summary report.

res.request object should be created as it is removed form v8 and it can be used for summary report.